controls/V-41818.rb

```diff
@@ -14,31 +16,40 @@
 only accept web server-generated session IDs and to create new session IDs once
 a user is authenticated will limit session hijacking.
   "
+  
+  desc  "check", "Review the web server documentation and deployed configuration 
+  to determine whether the web server accepts session IDs that are not system-generated.
+
+  If it is determined that the web server is not required to perform session management, 
+  this check is Not Applicable. 
+
+  NGINX web server versions after 1.11.0 have the $request_id embedded variable by default. 
+  This variable is a unique request identifier generated from 16 random bytes, in hexadecimal. 
+
+  Execute the following command to get the current version of NGINX running:
+    # nginx -v
+
+  If the current version of NGINX running is 1.11.0 or earlier, this is a finding. 
+  "
+  desc  "fix", "Upgrade to the lastest stable version of NGINX web server to use the '$request_id' 
+  embedded variable for generating unique identifers."
+
   impact 0.5
+  tag "severity": "medium"
   tag "gtitle": "SRG-APP-000223-WSR-000145"
   tag "gid": "V-41818"
   tag "rid": "SV-54395r3_rule"
   tag "stig_id": "SRG-APP-000223-WSR-000145"
   tag "fix_id": "F-47277r3_fix"
   tag "cci": ["CCI-001664"]
   tag "nist": ["SC-23 (3)", "Rev_4"]
-  tag "false_negatives": nil
-  tag "false_positives": nil
-  tag "documentable": false
-  tag "mitigations": nil
-  tag "severity_override_guidance": false
-  tag "potential_impacts": nil
-  tag "third_party_tools": nil
-  tag "mitigation_controls": nil
-  tag "responsibility": nil
-  tag "ia_controls": nil
-  tag "check": "Review the web server documentation and deployed configuration
-to determine whether the web server accepts session IDs that are not
-system-generated.
-
-If the web server does accept non-system-generated session IDs, this is a
-finding."
-  tag "fix": "Configure the web server to only accept session IDs that are
-created by the web server."
+
+# NGINX versions after 1.11.0 have the $request_id embedded variable by default
+# This variable is a unique request identifier generated from 16 random bytes, in hexadecimal
+
+  describe nginx do
+    its('version') { should cmp > '1.11.0' }
```

Comment:
use the Nginx version input here - 

WHY ARE we checking the NGINX version more than ONCE??